app/http.php

```diff
@@ -0,0 +1,1039 @@
+             */
+            $sourceDevice = getStorageDevice("/");
+            $localDevice = new Local();
+            $buffer = $sourceDevice->read($source);
+            if (!$localDevice->write($tmpSource, $buffer)) {
+                throw new Exception('Failed to copy source code to temporary directory', 500);
+            };
+
+            /**
+             * Create the mount folder
+             */
+            if (!\file_exists(\dirname($tmpBuild))) {
+                if (!@\mkdir(\dirname($tmpBuild), 0755, true)) {
+                    throw new Exception("Failed to create temporary directory", 500);
+                }
+            }
+
+            /**
+             * Create container
+             */
+            $variables = \array_merge($variables, [
+                'INTERNAL_RUNTIME_KEY' => $secret,
+                'INTERNAL_RUNTIME_ENTRYPOINT' => $entrypoint,
+                'INERNAL_EXECUTOR_HOSTNAME' => System::getHostname()
+            ]);
+            $variables = array_map(fn ($v) => strval($v), $variables);
+            $orchestration
+                ->setCpus((int) App::getEnv('OPEN_RUNTIMES_EXECUTOR_CPUS', '0'))
+                ->setMemory((int) App::getEnv('OPEN_RUNTIMES_EXECUTOR_MEMORY', '0'))
+                ->setSwap((int) App::getEnv('OPEN_RUNTIMES_EXECUTOR_MEMORY_SWAP', '0'));
+
+            /** Keep the container alive if we have commands to be executed */
+            $entrypoint = !empty($commands) ? [
+                'tail',
+                '-f',
+                '/dev/null'
+            ] : [];
+
+            $containerId = $orchestration->run(
+                image: $image,
+                name: $runtimeId,
+                hostname: $runtimeHostname,
+                vars: $variables,
+                command: $entrypoint,
+                labels: [
+                    'openruntimes-executor' => System::getHostname()
+                ],
+                workdir: $workdir,
+                volumes: [
+                    \dirname($tmpSource) . ':/tmp:rw',
+                    \dirname($tmpBuild) . ':/usr/code:rw'
+                ]
+            );
+
+            if (empty($containerId)) {
+                throw new Exception('Failed to create build container', 500);
+            }
+
+            $orchestration->networkConnect($runtimeId, App::getEnv('OPEN_RUNTIMES_EXECUTOR_NETWORK', 'executor_runtimes'));
+
+            /**
+             * Execute any commands if they were provided
+             */
+            if (!empty($commands)) {
+                $status = $orchestration->execute(
+                    name: $runtimeId,
+                    command: $commands,
+                    stdout: $stdout,
+                    stderr: $stderr,
+                    timeout: App::getEnv('OPEN_RUNTIMES_EXECUTOR_BUILD_TIMEOUT', 900)
+                );
+
+                if (!$status) {
+                    throw new Exception('Failed to build dependenices ' . $stderr, 500);
```

Comment:
This is coupled with our build workflow. In other runtimes we might not need dependencies, or the creation process might include other stuff.
```suggestion
                    throw new Exception('Failed to create runtime ' . $stderr, 500);
```

Reply:
Message updated